invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ManageDefaultBrandingAction/ManageDefaultBrandingAction.js

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2023 CERN.
+//
+// Invenio App RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import { Button, Message, Popup } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { http } from "react-invenio-forms";
+import PropTypes from "prop-types";
+
+export class ManageDefaultBrandingAction extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      error: undefined,
+    };
+  }
+
+  handleSetBranding = async (value) => {
+    const { recordCommunityEndpoint, updateRecordCallback } = this.props;
+    this.setState({ loading: true });
+    try {
+      const response = await http.put(recordCommunityEndpoint, {
+        default: value,
+      });
+
+      this.setState({ loading: false });
+      updateRecordCallback(response.data);
+    } catch (e) {
+      console.error(e);
+      this.setState({
+        error: e?.response?.data?.message || e?.message,
+        loading: false,
+      });
+    }
+  };
+
+  render() {
+    const { loading, error } = this.state;
+    const { result, isCommunityDefault } = this.props;
+
+    const communityTitle = result.metadata.title;
+    return error ? (
+      <Message
+        negative
+        floated="right"
+        className="community-branding-error"
+        size="tiny"
+      >
+        {error}
+      </Message>
+    ) : isCommunityDefault ? (
+      <Popup
+        trigger={
+          <Button
+            size="tiny"
+            labelPosition="left"
+            icon="paint brush"
+            floated="right"
+            onClick={() => this.handleSetBranding("")}
+            content={i18next.t("Remove branding")}
+            aria-label={i18next.t(
+              "{{communityTitle}} is a default branding for this record",
+              {
+                communityTitle,
+              }
+            )}
+            loading={loading}
+          />
+        }
+        content={i18next.t(
+          "Remove this community from appearing on top of the record details page."
+        )}
+        position="top center"
+      />
+    ) : (
+      <Popup
+        trigger={
+          <Button
+            size="tiny"
+            labelPosition="left"
+            icon="paint brush"
+            floated="right"
+            onClick={() => this.handleSetBranding(result.id)}
+            content={i18next.t("Set branding")}
+            aria-label={i18next.t(
+              "Set {{communityTitle}} as a default branding for this record",
+              {
+                communityTitle,
+              }
+            )}
+            loading={loading}
+          />
+        }
+        content={i18next.t(
+          "Set this community to appear on the top of the record details page."
+        )}
+        position="top center"
+      />
+    );
+  }
+}
```

Comment:
the implementation with double ternary operators is way too complex in terms of code readability, could you simplify it?

Reply:
Moved stuff around, should be more readable now. Let me know if it's still not very clear!